### DISCUSSION TITLE
I'm making Blocks+ extension with @Noname_awa

### DISCUSSION THREAD

**Discussion by MA-master-sec** [TurboWarp/extensions#640] [General]
### It will directly upgrade the original Scratch blocks and add some new features！
However, it still cannot be imported into the Scratch website

**Comment by LilyMakesThings** [comment]
Hearing this I sincerely hope it's not a "mega extension" that adds an absurd number of blocks. Those never work well 😕

**Reply by LilyMakesThings** [reply]
I was working on something I think might be similar to this called TurboCharged, that would add some miscellaneous blocks to every category by overriding them. However I didn't finish it.
![image-17](https://github.com/TurboWarp/extensions/assets/127533508/b967836d-d944-4341-bb21-62ae6a0213db)

**Reply by MA-master-sec** [reply]
In fact,It's not require adding blocks of unsupported types such as C-blocks.
I really want it to be a "big extension"
Begin from "motion" and end to "variables"

(Just join us)

**Reply by LilyMakesThings** [reply]
I will not "join you"

**Reply by yuri-kiss** [reply]
how :?

**Reply by MA-master-sec** [reply]
It's up to you
anyway,I want to use some of the ready-made extensions,including yours

**Reply by MA-master-sec** [reply]
> how :?

"how" what

**Reply by yuri-kiss** [reply]
how this: (https://github.com/TurboWarp/extensions/discussions/640#discussioncomment-6280975)

**Reply by yuri-kiss** [reply]
override the default blocks.

**Reply by Flappy25** [reply]
> override the default blocks.

I think you either modify the scratch vm

**Reply by yuri-kiss** [reply]
that it?

**Reply by Flappy25** [reply]
> that it?

Probably

**Reply by yuri-kiss** [reply]
vm.runtime.ext_scratch3_motion.runtime.addAddonBlock() could work right?

**Reply by yuri-kiss** [reply]
(used motion for the example)

**Reply by Flappy25** [reply]
> vm.runtime.ext_scratch3_motion.runtime.addAddonBlock() could work right?

Possibly

**Reply by yuri-kiss** [reply]
how do you refresh the block pallete?

**Reply by Flappy25** [reply]
> how do you refresh the block pallete?

I don't know, I'm pretty new to this

**Reply by yuri-kiss** [reply]
i figured it out :P, its `vm.runtime.extensionManager.refreshBlocks()`

**Reply by David-Orangemoon** [reply]
please don't use any of mine either.

**Reply by MA-master-sec** [reply]
Just duplicate with the original id

**Comment by LilyMakesThings** [comment]
> It's up to you anyway,I want to use some of the ready-made extensions,including yours

Please DO NOT use my extensions as a reference to fill your mega extension. If you want to use those blocks, use the extension.

**Reply by MA-master-sec** [reply]
Just joking : )

**Comment by ghost** [comment]
This sounds pretty cool

**Reply by LilyMakesThings** [reply]
Taking other people's extensions and putting them into 1 extension is not "cool"

**Reply by ghost** [reply]
As long as he/she adds different blocks, I think it's OK.

**Comment by MA-master-sec** [comment]
# It will NOT support all languages

**Reply by MA-master-sec** [reply]
He only has some commonly used languages such as English, Chinese, French, Russian, Japanese, etc

**Reply by LilyMakesThings** [reply]
...this also applies to every extension on TurboWarp right now
